Complete task 3

Krekep opened the pull request:
# Задача 3. Построение недетерминированного конечного автомата по графу

* **Мягкий дедлайн**: 19.09.2021, 23:59
* **Жёсткий дедлайн**: 22.09.2021, 23:59
* Полный балл: 2

## Задача

- [x] Используя возможности [pyformlang](https://pyformlang.readthedocs.io/en/latest/) реализовать **функцию** построения недетерминированного конечного автомата по [графу](https://networkx.org/documentation/stable/reference/classes/multidigraph.html), в том числе по любому из графов, которые можно получить, пользуясь функциональностью, реализованной в [Задаче 1](https://github.com/JetBrains-Research/formal-lang-course/blob/main/tasks/task1.md) (загруженный из датасета по имени граф, сгенерированный синтетический граф). Предусмотреть возможность указывать стартовые и финальные вершины. Если они не указаны, то считать все вершины стартовыми и финальными.
- [x] Добавить необходимые тесты.

Krekep committed:
[Codestyle] Rewrite console

- Rewrite parser for input

Krekep committed:
[Project] Add task 3

- Add console command for translate graph to nfa
- Add method graph_to_nfa
- Add tests for graph_to_nfa
- Update project/readme.md

Krekep committed:
[Codestyle] Fix codestyle

vadyushkins reviewed (changes requested):
В целом всё хорошо :+1: 
Только добавьте, пожалуйста, больше разных графов (граф без вершин, граф на 1 вершине, граф без ребер, т.п.) :eyes: 
---
- [ ] Также стоит проверять стартовые и конечные вершины на валидность

Krekep committed:
[Fix] Add new tests

- Fix start and final states
- Add start and finish vertices check
- Add new tests

vadyushkins commented:
Предлагаю уменьшить ДКА в `test_get_min_dfa`. Так как `pyformlang` недетерминированным образом строит мин ДКА.

Без пройденного CI сложно засчитывать задачу :)

Krekep committed:
[Fix] Fix min dfa test

Krekep commented:
Этот тест через раз проходил, при чём на он одновременно проходил и не проходил на пуше и пулл реквесте :(

vadyushkins reviewed (approved):
Зачтено :+1:
